src/runtime/omp/omp_allocator.cpp

```diff
@@ -33,6 +33,29 @@
 namespace hipsycl {
 namespace rt {
 
+namespace {
+/// borrowed from LLVM
+/// Returns the next power of two (in 64-bits) that is strictly greater than A.
+/// Returns zero on overflow.
+uint64_t NextPowerOf2(uint64_t A) {
+  A |= (A >> 1);
+  A |= (A >> 2);
+  A |= (A >> 4);
+  A |= (A >> 8);
+  A |= (A >> 16);
+  A |= (A >> 32);
+  return A + 1;
+}
+
+/// Returns the power of two which is greater than or equal to the given value.
+/// Essentially, it is a ceil operation across the domain of powers of two.
+uint64_t PowerOf2Ceil(uint64_t A) {
+  if (!A)
+    return 0;
+  return NextPowerOf2(A - 1);
```

Comment:
Since this is such a small section of code, would it make sense to adapt it to the style of the surrounding code? Or maybe move it to the runtime's `util.hpp` as it might be useful in other places as well?